addons/binding/org.openhab.binding.km200/src/main/java/org/openhab/binding/km200/internal/KM200Device.java

```diff
@@ -0,0 +1,385 @@
+/**
+ * Copyright (c) 2010-2018 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.km200.internal;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+
+import javax.xml.bind.DatatypeConverter;
+
+import org.apache.commons.lang.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParseException;
+import com.google.gson.JsonParser;
+
+/**
+ * The KM200Device representing the device with its all capabilities
+ *
+ * @author Markus Eckhardt - Initial contribution
+ *
+ */
+
+public class KM200Device {
+
+    private final Logger logger = LoggerFactory.getLogger(KM200Device.class);
+    private final JsonParser jsonParser = new JsonParser();
+    private final KM200Cryption comCryption;
+    private final KM200Comm<KM200Device> deviceCommunicator;
+
+    /* valid IPv4 address of the KMxxx. */
+    protected String ip4Address;
+
+    /* The gateway password which is provided on the type sign of the KMxxx. */
+    protected String gatewayPassword;
+
+    /* The private password which has been defined by the user via EasyControl. */
+    protected String privatePassword;
+
+    /* The returned device charset for communication */
+    protected String charSet;
+
+    /* Needed keys for the communication */
+    protected byte[] cryptKeyInit;
+    protected byte[] cryptKeyPriv;
+
+    /* Buderus_MD5Salt */
+    protected byte[] md5Salt;
+
+    /* Device services */
+    public HashMap<String, KM200ServiceObject> serviceTreeMap;
```

Comment:
> Declare as Map / List etc check everywhere.

This still seems present, but its not immediately blocking.

Reply:
All should be changed now.